linfa-bayes/src/error.rs

```diff
@@ -0,0 +1,46 @@
+use std::fmt;
+
+use ndarray::ShapeError;
+use ndarray_stats::errors::MinMaxError;
+
+pub type Result<T> = std::result::Result<T, BayesError>;
+
+#[derive(Debug)]
+pub enum BayesError {
```

Comment:
there is now a global error type for algorithms, which you can extend https://github.com/rust-ml/linfa/blob/master/src/error.rs

Reply:
Some errors like `BayesError::Priors` are very specific to the algorithm, do you still want to move them over to the root?

Reply:
no I changed my mind on this. We should use an error library which offers extensible errors. For the time being just use your own error type here